integrationtests/tests/fixtures/mod.rs

```diff
@@ -832,6 +851,35 @@ impl FederationTest {
         }
     }
 
+    /// Removes the ecash nonces from the fed DB to simulate the fed losing track of what ecash
+    /// has already been spent
+    pub async fn clear_ecash_nonces(&self) {
+        for server in &self.servers {
+            block_on(async {
+                let svr = server.borrow_mut();
+                let mut dbtx = svr.database.begin_transaction().await;
+
+                {
+                    let mut module_dbtx =
+                        dbtx.with_module_prefix(LEGACY_HARDCODED_INSTANCE_ID_MINT);
+
+                    let nonces: Vec<NonceKey> = module_dbtx
+                        .find_by_prefix(&NonceKeyPrefix)
```

Comment:
BTW. Should we rename this key to `SpentNonceKeyPrefix` (it's a critical distinction, and I think it would make understanding code much easier @m1sterc001guy )

Reply:
I'd be down for that in a separate PR